cgo compilation fails with older kernel sources and go 1.24

Go 1.24 includes a [patch](https://github.com/golang/go/commit/65679cfeb4b2fa0f24ac4ed8757b8a83ab0d5690) to re-enable the fallback to `/dev/urandom` when the random syscall isn't available.  This is great for teams who still have customers who are putting off updating to an OS that's still within its support window.

The problem is that it also introduced a `crypto/rand/internal/seccomp` package in the standard library, which relies on `SYS_getrandom` and `SYS_seccomp` being defined in `sys/syscall.h`, leading to compilation errors like
```
.../src/crypto/internal/sysrand/internal/seccomp/seccomp_linux.go:58:45: error: use of undeclared identifier 'SYS_getrandom'
   58 |         BPF_JUMP(BPF_JMP | BPF_JEQ | BPF_K, SYS_getrandom, 0, 1),
      |                                             ^
.../src/crypto/internal/sysrand/internal/seccomp/seccomp_linux.go:63:22: error: invalid application of 'sizeof' to an incomplete type 'struct sock_filter[]'
   63 |         .len = sizeof(filter) / sizeof((filter)[0]),
      |                      ^~~~~~~~
.../src/crypto/internal/sysrand/internal/seccomp/seccomp_linux.go:66:17: error: use of undeclared identifier 'SYS_seccomp'
   66 |     if (syscall(SYS_seccomp, SECCOMP_SET_MODE_FILTER, 0, &prog)) {
      |                 ^
```
if you are compiling on a system with kernel headers older than 3.17[^1].

[^1]: We're not building on an out-of-support OS, but we are using a custom sysroot to ensure that the build outputs can _run_ on such an OS.

That is an `internal` package, and it's only used by a few unit tests within `crypto/rand`, so `go build` won't actually try to compile it unless you're trying to compile the standard library unit tests.  Unfortunately, because `rules_go` treats the standard library as monolithic dependency, it _does_ try to compile it, and so it fails.

The fact that this is a bazel-only problem makes me think it's unlikely that there would be interest in fixing this issue upstream in the go sources.  Probably the simplest fix would be to simply exclude `src/crypto/internal/sysrand/internal/seccomp` from the SDK glob.  There seems to be precident for that sort of thing:
https://github.com/bazel-contrib/rules_go/blob/eef92f5c9743277a3d84e8e6b478d957ac679a67/go/private/BUILD.sdk.bazel#L24-L41

There are probably other examples of `internal` packages within the standard library which are only ever used by unit tests; finding them comprehensively would be nice but not easy in general.

--- a/go/private/BUILD.sdk.bazel
+++ b/go/private/BUILD.sdk.bazel
@@ -29,6 +29,8 @@ filegroup(
             "src/**/*_test.go",
             "src/**/testdata/**",
             "src/cmd/**",
+            # Only used by tests, cgo fails with linux before 3.17
+            "src/crypto/internal/sysrand/internal/seccomp/**",
             "src/log/slog/internal/benchmarks/**",
             "src/log/slog/internal/slogtest/**",
             "src/internal/obscuretestdata/**",

